src/arx7.js

```diff
@@ -1,133 +1,199 @@
+      log(`Unrecognized command`);
+      this.client.say(from, `Not enough commands. ${usage}`);
+      return;
+    }
 
-    if (key != null) {
-      log(`${message.args[1]} is +k. Rejoining with password.`);
-      client.join(`${message.args[1]} ${key}`);
-      droppedChannels.push(message.args[1]);
+    // Verify Password
+    if (args[3] !== this.config.adminPassword) {
+      log(`Invalid Admin Password`);
+      this.client.say(from, `Invalid password. ${usage}`);
+      return;
     }
-    else {
-      log(`${message.args[1]} is +k. No key found. Skipping channel`);
+
+    let channel_index = this.channels.indexOf(args[2].toLowerCase());
+
+    // Verify Channel Existence
+    if (!this.config.channels[channel_index]) {
+      log(`Invalid channel`);
+      this.client.say(from, `Invalid channel. ${usage}`);
+      return;
+    }
+
+    // Search and Verify Plugin Existence
+    let exists = this.commands.some(c => {
+      return c.constructor.name.toLowerCase() === args[1].toLowerCase();
+    });
+
+    if (!exists) {
+      log(`Invalid plugin`);
+      this.client.say(from, `Invalid plugin. ${usage}`);
+      return;
+    }
+
+    // Add Plugin
+    if (args[0].toLowerCase() === 'add') {
+      log(`ENABLE command for ${args[1]} in ${args[2]}`);
+      let plugin = args[1].toLowerCase();
+      let idx = this.config.channels[channel_index].plugins.indexOf(plugin);
+
+      if (idx > -1) {
+        this.client.say(from, `Plugin ${args[1]} already enabled.`);
+      }
+      else {
+        log(`Enabling ${args[1]} for ${args[2]}`);
+        this.config.channels[channel_index].plugins.push(plugin);
+        this.client.say(from, `Enabled ${args[1]} for ${args[2]}`);
+      }
+    }
+
+    // Remove Plugin
+    if (args[0].toLowerCase() === 'remove') {
+      let plugin = args[1].toLowerCase();
+      let idx = this.config.channels[channel_index].plugins.indexOf(plugin);
+      this.config.channels[channel_index].plugins.splice(idx, 1);
+      log(`DISABLE command for ${args[1]} in ${args[2]}`);
+      this.client.say(from, `Disabled ${args[1]} for ${args[2]}`);
     }
   }
-  // Log other errors
-  else {
-    log(`ERROR: ${message.command}`);
+
+  kick(channel, nick, by, reason, message) {
+    if (nick === this.client.nick) {
+      log(`Kicked from ${channel} by ${by}: ${reason}`);
+      setTimeout(() => {
+        log(`Rejoining ${channel}`);
+        this.client.join(channel);
+      }, 1000 * 3);
+    }
+  }
+
+  error(message) {
+    if (message.command == 'err_bannedfromchan') {
+      log(`Banned from ${message.args[1]}. Rejoining in in 3 minutes`);
+
+      // `()=>` ensures there is a delay; otherwise it continuously fires
+      setTimeout(() => this.client.join(message.args[1]), 1000 * 60 * 3);
+    }
+    // Attempt to rejoin +k channels correctly
+    else if (message.command == 'err_badchannelkey') {
+      // Prevent infinite rejoin attempts
+      if (this.droppedChannels.includes(message.args[1])) {
```

Comment:
Not a new bug, but I just realized if you kick the bot from a channel with +k twice, it won't join the second time. Is it possible to remove a channel from the list when the bot successfully joins a channel?